services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/volumes_resolver.py

```diff
@@ -82,28 +85,63 @@ def _volume_name(cls, path: Path) -> str:
         return f"{path}".replace(os.sep, "_")
 
     @classmethod
-    def source(cls, compose_namespace: str, path: Path) -> str:
-        return f"{compose_namespace}{cls._volume_name(path)}"
+    def source(cls, path: Path, node_uuid: NodeID, run_id: UUID) -> str:
+        """
+        Guarantees that the volume name is unique between runs while also
+        taking into consideration the limit for the volume name's length
+        (255 characters).
+
+        Example:
+        >>> source(
+            path="/this/is/a/data/folder",
+            node_uuid="17f5b46d-505c-489e-9818-f6d294c892d9",
+            run_id="17f5b46d-505c-489e-9818-f6d294c892d9"
+        )
+        dyv_17f5b46d-505c-489e-9818-f6d294c892d9_this_is_a_data_folder_17f5b46d-505c-489e-9818-f6d294c892d9
+        """
+        # NOTE: this name is only used when removing the volume.
+        # NOTE: issues can occur when the paths of the mounted outputs, inputs
+        # and state folders are very long and share the same subdirectory path.
+        # If this becomes on issue, inverting the `volume_name`, appended after
+        # the `run_id` will avoid collisions.
+        full_unique_name = f"{DY_SIDECAR_NAMED_VOLUME_PREFIX}_{run_id}{cls._volume_name(path)}_{node_uuid}"
+        return full_unique_name[:255]
 
     @classmethod
     def mount_entry(
-        cls,
-        swarm_stack_name: str,
-        compose_namespace: str,
-        path: Path,
-        node_uuid: NodeID,
-        run_id: UUID,
-    ) -> Dict[str, Any]:
+        cls, swarm_stack_name: str, path: Path, node_uuid: NodeID, run_id: UUID
+    ) -> dict[str, Any]:
         """
         mounts local directories form the host where the service
         dynamic-sidecar) is running.
         """
         return {
+            "Source": cls.source(path, node_uuid, run_id),
             "Target": cls.target(path),
             "Type": "volume",
             "VolumeOptions": {
                 "Labels": {
-                    "source": cls.source(compose_namespace, path),
+                    "source": cls.source(path, node_uuid, run_id),
```

Comment:
can we have the set of "standard" ids on the volumes too? at least it helps to find to which service, study, user they are attached to

Reply:
I've added them.